src/FriendPanel.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useRef } from 'react';
+import { flushSync } from 'react-dom';
+import './FriendPanel.css';
+
+export default function FriendPanel({selectFriend, selectedFriend, message, setMessage, history, setHistory}) {
+    const [friends, setFriends] = useState([]);
+    const [newFriend, setNewFriend] = useState('');
+    const selectedRef = useRef(null);
+  
+    const handleChangeFriend = (event) => {
+      setNewFriend(event.target.value);
+    };
+  
+    function addFriend() {
+      if (friends.includes(newFriend)) {
+        flushSync(() => {selectFriend(newFriend)});
+        selectedRef.current.scrollIntoView();
+        console.log('user already exists');
+        console.log(selectedRef.current);
+      } else if(newFriend === '') {
+        console.log('no user entered');
+      } else {
+        const nextFriends = [...friends, newFriend];
+        flushSync(() => {
+            setFriends(nextFriends);
+            selectFriend(newFriend);
```

Comment:
This is one of the examples that `useReducer` can be useful to group states together.